### DISCUSSION TITLE
prisma migration with plsql

### DISCUSSION THREAD

**Discussion by avadhut0110** [prisma/prisma#22283] [Q&A] [answered]
### Question

Hello ,
I am using prisma with posgreSql when i make some changes in schema.prisma I use prisma migration command. in case of making changes in exsisting table when I use same command it shows data will be lost.
what I am doing wrong?
command i use  **npx prisma migrate dev**  

![Screenshot 2023-12-06 151230](https://github.com/prisma/prisma/assets/153074253/2f78d73d-60ed-4721-a8f1-6503f4c2062a)

### Expected behavior (optional)

the changes in schema.prisma should reflect in database and data should not be lost

### Information about Prisma Schema, Client Queries and Environment (optional)

```prisma
// Add your schema.prisma
```

```ts
// Add any relevant Prisma Client queries here
```

- OS: 
- Database: 
- Node.js version: 

- Run `prisma -v` to see your Prisma version and paste it

**Comment by ludralph** [comment, marked as answer]
Hi @avadhut0110 👋 

Thank you for raising this question.

Prisma Migrate attempts to create a new migration based on the changes in your `schema.prisma` file and apply it to your development database. However, Prisma Migrate may determine that the changes cannot be applied without potentially losing data. This is why you might see a warning that data will be lost.

The migrate dev command will prompt you to reset the database in the following scenarios:

- Migration history conflicts caused by [modified or missing migrations](https://www.prisma.io/docs/concepts/components/prisma-migrate/migration-histories#do-not-edit-or-delete-migrations-that-have-been-applied)

- The database schema has drifted away from the end-state of the migration history

If you need to apply migrations to other environments (like production) where data loss is not acceptable, you should use `prisma migrate deploy`, which applies migrations without resetting the database.

**Reply by avadhut0110** [reply]
Thank you so much 
it worked.

**Comment by Akhileshpm** [comment]
Hi @avadhut0110, I found a workaround for this but it might not be a good way. I ended up in the same situation when I tried to change the foreign key of a table. Since the tables that I am dealing with had complicated relations, Prisma couldn't resolve it. So I tried removing records from the table and running the migrate dev command. It failed again, then I went through the migrations folder and found the current migrations script created over there(even though the migrate command failed), and also in the **prisma_migrations** in the database. I deleted both the appearances and tried prisma migrate dev once again and it worked. Hope this might help and there is no hidden problems.
